dotnet/src/InternalUtilities/planning/PlannerInstrumentation.cs

```diff
@@ -127,11 +128,21 @@ internal static partial class PlannerInstrumentation
         Message = "Plan created.")]
     static partial void LogPlanCreated(this ILogger logger);
 
-    [LoggerMessage(
-        EventId = 3,
-        Level = LogLevel.Trace, // Sensitive data, logging as trace, disabled by default
-        Message = "Plan:\n{Plan}")]
-    static partial void LogPlan(this ILogger logger, object plan);
+    private static readonly Action<ILogger, string, Exception?> s_logPlan =
+        LoggerMessage.Define<string>(
+            logLevel: LogLevel.Trace,   // Sensitive data, logging as trace, disabled by default
+            eventId: 3,
+            "Plan:\n{Plan}");
```

Comment:
Can we log the goal here too? Helps gauge expected behavior

Reply:
The goal is logged before the planner is invoked.